src/mumble/AudioOutput.cpp

```diff
@@ -128,8 +133,17 @@ float AudioOutput::calcGain(float dotproduct, float distance) {
 }
 
 void AudioOutput::wipe() {
-	foreach (AudioOutputUser *aop, qmOutputs)
-		removeBuffer(aop);
+	QList< AudioOutputBuffer * > list;
+
+	qrwlOutputs.lockForRead();
+	for (AudioOutputBuffer *buffer : qmOutputs) {
+		list.append(buffer);
+	}
+	qrwlOutputs.unlock();
+
+	for (AudioOutputBuffer *buffer : list) {
+		removeBuffer(buffer);
+	}
```

Comment:
There should be a comment explaining why this is done in such a seemingly funny way.